### ISSUE TITLE
Can we add parameters/secrets while pushing the kernels ? 

### ISSUE THREAD

**Issue by bleakprestiger** [Kaggle/kaggle-api#582]

**Comment by stevemessick** [comment]
Could you give us a little more detail? An example of what you want to do would be great.

**Comment by YuenSzeHong** [comment]
> Could you give us a little more detail? An example of what you want to do would be great.

I guess he ment user secrets (like api keys that we dont want to expose)

**Comment by stevemessick** [comment]
No, I do not think we want to do that. I think it would introduce security issues. @rosbo Could you confirm?

The way to manage your API key is with the `kaggle_secrets` package. See:
https://www.kaggle.com/code/harupy/how-to-use-kaggle-api-as-a-python-library

This package is only available in Kaggle notebooks and is a secure way to set up user name and key.

**Comment by rosbo** [comment]
@stevemessick To clarify, `kaggle_secrets` is used to access secrets in a notebook (i.e. get the secret value).

```
from kaggle_secrets import UserSecretsClient

secret_value = UserSecretsClient().get_secret("YOUR_SECRET_NAME")
```

`kaggle_secrets` doesn't allow setting the value, it only reads the value. Currently, the value must be set from `Notebook Editor > Add-ons > Secrets`. Another thing to note is that secrets are set at your account level. Then, you can decide which secrets you want to attach to a specific notebook. If you update the value of a secret, the new value will be used by all notebooks using that secret.

@bleakprestiger Are you asking for an option to "attach" a given list of secrets automatically when calling `kaggle kernels push`?

Or are you asking for new methods in the CLI to set user secrets. Something along the lines of:

```
kaggle secrets set MY_API_KEY abcdefghijk123
```

Thank you

**Comment by bleakprestiger** [comment]
yes @rosbo, can we attach a given list of secrets when pushing the kernels ? Also can we pass some input variables while pushing the kernels ?

**Comment by gspitz01** [comment]
I would also like to express my support for some way to add a list of secrets to a kernel when using `kaggle kernels push`, perhaps as part of kernel-metadata.json.

**Comment by ternaus** [comment]
A concrete automation use case: a training launcher prepares source and data inputs, then runs kaggle kernels push for a private kernel. Its generated entry point reads an existing WANDB_API_KEY only by label through UserSecretsClient. The launcher cannot attach that label to the newly pushed kernel, so someone must open the Notebook editor before the run can authenticate.

The current CLI documents this gap, and its generated save-kernel request has no field for secret labels. PR #944 only added argparse options; it did not pass them to kernels_push_cli or the request, and it closed unmerged after the CLA check failed.

The smallest useful contract is an authenticated backend/API operation that accepts existing secret labels only, attaches them server-side for a new or existing private kernel, and never puts plaintext values in metadata, CLI output, or the pushed source. A focused integration test should cover a push whose code retrieves a previously created secret by label. If that backend operation already exists but is absent from the public API spec, exposing it in the spec is the first step.
